### ISSUE TITLE
Certified copies: As a System Admin, I want to control when a certified copy template can be printed and issued

### ISSUE THREAD

**Issue by euanmillar** [opencrvs/opencrvs-core#7585]
### Description

Previously, it was only possible for one certificate SVG template type to be issued for each vital event on a 1:1 basis.  1 birth certificate, 1 death certificate etc. 
In this feature request: https://github.com/opencrvs/opencrvs-core/issues/7584 we have made it possible to support n number of SVG templates for each event.

When a user prints a registration, they can now select which certificate template they wish to print.

![Image](https://github.com/user-attachments/assets/12d6d4ab-0d31-49bd-98d8-b38852aea1fe)

All the certificates which can be printed are served from the countryconfig microservice [in a JSON configuration that is defined here](https://github.com/opencrvs/opencrvs-countryconfig/blob/5db618f87a78e3f93830be61a8d2b3a6691ab23f/src/api/certificates/handler.ts#L43)

In some countries, the certificates which should be options that are displayed to the user depend on information in the declaration draft.  For example:

- If the event occurred within 12 months or not. Therefore logic needs to look at data within the draft (date of birth) before allowing the certificate option to be visible.  In Somalia, I want the system to only show me the appropriate short-form or long-form certificate based on the child's date of birth when printing, so that the correct certificate is issued, reducing errors or possible delays.
- Some certificates should only be printed once.  Some certificates should not be printed first. Therefore logic needs to look at data within the draft history, to determine if certificates have been printed already, and specifically which certificate types may have been printed already.

<img width="720" alt="Image" src="https://github.com/user-attachments/assets/956165c4-73dd-4426-9d1f-440c0e6072b5" />

Certificate options can be shown / hidden based on:
- If the event occurred within a period of time. Therefore logic needs to look at data within the draft (date of birth) before allowing the certificate option to be visible.
- Some certificates should only be printed once.  Some certificates should not be printed first. Therefore logic needs to look at data within the draft history, to determine if certificates have been printed already, and specifically which certificate types may have been printed already.
- Some certificates may only be able to be printed by certain roles
- Whichever certificate template is selected, the selected template needs tracked in the backend to enable the Metabase analytics in subsequent tickets

We will never know all the conditionals that may exist for all countries, so we want to add configurable conditional logic to the JSON configuration for each certificate.  In this way countries can decide how they want the select options to behave.

We should define some JSON configuration naming conventions that translate to a set of supported functions hardcoded in core that satisfy the use cases above, or perhaps a hybrid approach that uses the eval functionality we are used to in form field conditionals.

**Comment by rikukissa** [comment]
I think some further evaluation on the alternatives for the JS eval is in place here. I think this is a rather common problem in API design and I’m sure there are examples we could look at. The reason I’m saying this is in the long term we want to be moving away from eval and introducing new APIs that use it of course makes this more difficult. The reason why it’s difficult is the exposed layer is not very well contained. 

The eval expression executes within the context of the form where access to Redux application state properties such as the form: “values”, application config: “offlineCountryConfig”, draft “draftData” and the logged in user: “userDetails” are available to it.

These variables are deprecated so if we choose to implement it the eval way, let’s use the new variables [here](https://github.com/opencrvs/opencrvs-core/blob/develop/packages/client/src/forms/utils.ts#L587-L593)

```
$form
$config
$draft
$user
```

The API design principle in this case is the fewer things we need to expose outside the form data, the better. 

@euanmillar

**Comment by euanmillar** [comment]
@rikukissa I think that we should put this on hold in IET until @tareq89 has completed the collector form work.  Then perhaps we can use something like the conditional FieldAPI you have been working on?

**Comment by rikukissa** [comment]
Makes sense. If this is needed earlier however, there is a relatively easy and aligned way of doing it on the old side as well.

**Comment by euanmillar** [comment]
Madagascar's versioning requirements might be satisfied by this requirement: https://github.com/opencrvs/opencrvs-core/discussions/9593
